### ISSUE TITLE
:py not working on Mac OS Yosemite

### ISSUE THREAD

**Issue by SirVer** [neovim/neovim#1987]
I did 

~~~bash 
$ brew tap neovim/homebrew-neovim
$ brew reinstall --HEAD neovim
$ pip install -U neovim
~~~

Then nvim gave tons of errors, so I tried running `pynvim`:

~~~
Traceback (most recent call last):
  File "/usr/local/bin/pynvim", line 9, in <module>
    load_entry_point('neovim==0.0.27', 'console_scripts', 'pynvim')()
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 610, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/neovim/ui/cli.py", line 66, in main
    from .tickit_ui import TickitUI
  File "/usr/local/lib/python2.7/site-packages/neovim/ui/tickit_ui.py", line 7, in <module>
    import cffi
ImportError: No module named cffi

~~~

So installed that, but still:

~~~
$ pynvim
Package tickit was not found in the pkg-config search path.
Perhaps you should add the directory containing `tickit.pc'
to the PKG_CONFIG_PATH environment variable
No package 'tickit' found
Traceback (most recent call last):
  File "/usr/local/bin/pynvim", line 9, in <module>
    load_entry_point('neovim==0.0.27', 'console_scripts', 'pynvim')()
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 610, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python2.7/site-packages/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/neovim/ui/cli.py", line 66, in main
    from .tickit_ui import TickitUI
  File "/usr/local/lib/python2.7/site-packages/neovim/ui/tickit_ui.py", line 190, in <module>
    ''', libraries=['tickit']).load_library()
  File "/usr/local/lib/python2.7/site-packages/cffi/verifier.py", line 74, in load_library
    self._compile_module()
  File "/usr/local/lib/python2.7/site-packages/cffi/verifier.py", line 139, in _compile_module
    outputfilename = ffiplatform.compile(tmpdir, self.get_extension())
  File "/usr/local/lib/python2.7/site-packages/neovim/ui/tickit_ui.py", line 170, in get_extension
    'tickit']).strip()[2:]
  File "/usr/local/Cellar/python/2.7.7_2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '['pkg-config', '--cflags', 'tickit']' returned non-zero exit status 1
~~~

No idea what is wrong. This happens when I run nvim:

~~~
nvim -u NONE
:py import sys
Provider for method "python_execute" is not available
~~~

Any ideas?

**Comment by justinmk** [comment]
What was the output of `pip install --upgrade neovim`?

**Comment by SirVer** [comment]
~~~bash
$ pip install -U neovim                                       
Requirement already up-to-date: neovim in /usr/local/lib/python2.7/site-packages
Requirement already up-to-date: click>=3.0 in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already up-to-date: msgpack-python>=0.4.0 in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already up-to-date: trollius in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already up-to-date: greenlet in /usr/local/lib/python2.7/site-packages (from neovim)
Cleaning up...
~~~

uninstalling, reinstalling:
~~~~bash
pip install neovim                                           ~
Downloading/unpacking neovim
  Downloading neovim-0.0.27.tar.gz (77kB): 77kB downloaded
  Running setup.py (path:/private/var/folders/6l/4ytkxqrd2nq7shcwyslmr6vr0000gn/T/pip_build_sirver/neovim/setup.py) egg_info for package neovim

Requirement already satisfied (use --upgrade to upgrade): click>=3.0 in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already satisfied (use --upgrade to upgrade): msgpack-python>=0.4.0 in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already satisfied (use --upgrade to upgrade): trollius in /usr/local/lib/python2.7/site-packages (from neovim)
Requirement already satisfied (use --upgrade to upgrade): greenlet in /usr/local/lib/python2.7/site-packages (from neovim)
Installing collected packages: neovim
  Running setup.py install for neovim
    building 'neovim.ui.screen' extension
    clang -fno-strict-aliasing -fno-common -dynamic -I/usr/local/include -I/usr/local/opt/sqlite/include -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/usr/local/Cellar/python/2.7.7_2/Frameworks/Python.framework/Versions/2.7/include/python2.7 -c neovim/ui/screen.c -o build/temp.macosx-10.9-x86_64-2.7/neovim/ui/screen.o
    neovim/ui/screen.c:7916:28: warning: unused function '__Pyx_PyObject_AsString' [-Wunused-function]
    static CYTHON_INLINE char* __Pyx_PyObject_AsString(PyObject* o) {
                               ^
    neovim/ui/screen.c:7913:32: warning: unused function '__Pyx_PyUnicode_FromString' [-Wunused-function]
    static CYTHON_INLINE PyObject* __Pyx_PyUnicode_FromString(const char* c_str) {
                                   ^
    neovim/ui/screen.c:297:29: warning: unused function '__Pyx_Py_UNICODE_strlen' [-Wunused-function]
    static CYTHON_INLINE size_t __Pyx_Py_UNICODE_strlen(const Py_UNICODE *u)
                                ^
    neovim/ui/screen.c:8028:33: warning: unused function '__Pyx_PyIndex_AsSsize_t' [-Wunused-function]
    static CYTHON_INLINE Py_ssize_t __Pyx_PyIndex_AsSsize_t(PyObject* b) {
                                    ^
    neovim/ui/screen.c:8053:33: warning: unused function '__Pyx_PyInt_FromSize_t' [-Wunused-function]
    static CYTHON_INLINE PyObject * __Pyx_PyInt_FromSize_t(size_t ival) {
                                    ^
    neovim/ui/screen.c:6525:28: warning: unused function '__Pyx_CyFunction_InitDefaults' [-Wunused-function]
    static CYTHON_INLINE void *__Pyx_CyFunction_InitDefaults(PyObject *func, size_t size, int pyobjects) {
                               ^
    neovim/ui/screen.c:6534:27: warning: unused function '__Pyx_CyFunction_SetDefaultsTuple' [-Wunused-function]
    static CYTHON_INLINE void __Pyx_CyFunction_SetDefaultsTuple(PyObject *func, PyObject *tuple) {
                              ^
    neovim/ui/screen.c:6539:27: warning: unused function '__Pyx_CyFunction_SetDefaultsKwDict' [-Wunused-function]
    static CYTHON_INLINE void __Pyx_CyFunction_SetDefaultsKwDict(PyObject *func, PyObject *dict) {
                              ^
    neovim/ui/screen.c:6544:27: warning: unused function '__Pyx_CyFunction_SetAnnotationsDict' [-Wunused-function]
    static CYTHON_INLINE void __Pyx_CyFunction_SetAnnotationsDict(PyObject *func, PyObject *dict) {
                              ^
    neovim/ui/screen.c:6895:27: warning: function '__Pyx_PyInt_As_long' is not needed and will not be emitted [-Wunneeded-internal-declaration]
    static CYTHON_INLINE long __Pyx_PyInt_As_long(PyObject *x) {
                              ^
    neovim/ui/screen.c:6990:26: warning: function '__Pyx_PyInt_As_int' is not needed and will not be emitted [-Wunneeded-internal-declaration]
    static CYTHON_INLINE int __Pyx_PyInt_As_int(PyObject *x) {
                             ^
    11 warnings generated.
    clang -bundle -undefined dynamic_lookup -L/usr/local/lib -L/usr/local/opt/sqlite/lib build/temp.macosx-10.9-x86_64-2.7/neovim/ui/screen.o -o build/lib.macosx-10.9-x86_64-2.7/neovim/ui/screen.so

    Installing pynvim script to /usr/local/bin
Successfully installed neovim
Cleaning up...
~~~~

**Comment by tarruda** [comment]
@SirVer the `Package tickit was not found in the pkg-config search path` error can be fixed by installing [libtickit](https://github.com/neovim/libtickit) to some directory and starting pynvim with `PKG_CONFIG_PATH` set to `$LIBTICKIT_PREFIX/lib/pkgconfig`. Note that it is only required for the terminal UI, if you run `pynvim -g` it will try to start the gtk+ UI which requires python gobject bindings for cairo and gtk.

I don't have access to OSX so I can't help much with system configuration, but I've written a [script](https://gist.github.com/tarruda/b67472898a7e4bd81572) that downloads and installs the gtk bindings on ubuntu for a [pyenv](https://github.com/yyuu/pyenv) python installation, maybe you can adapt it for your needs.

**Comment by tarruda** [comment]
If you just want to try a custom nvim UI, [this](https://github.com/coolwanglu/neovim.as) may be easier to setup on OSX

**Comment by SirVer** [comment]
I returned to this today and it just started working after updating neovim from HEAD and reinstalling the neovim python client again. 

I also had the chance to try setting up https://github.com/carlosdcastillo/vim-mode which is painless to setup, but as buggy as to be unusable (it only claims pre-alpha status in the docs, that is accurate) and  https://github.com/coolwanglu/neovim.as which I was unable to get up and running - no idea how well it performs.

I installed all the gtk3 dependencies for pynvim and that started up just fine, but was buggy. For example `<c-h>` was mapped to backspace which is wrong outside of the terminal, also `<tab>` apparently did not reach neovim since it did not expand my UltiSnips snippets (though imap was set correctly). It also looked very wrong with fonts and layout being off. Summary: it starts up, but it is unusable for any work.

My conclusion is that it is too early to judge if the RPC based UI is working out for neovim or not. Right now, nothing seems to be usable yet. I do not really care about the console improvements that landed lately, I want my GUI with bells, whistles, FocusLost autocommands, undercurls.... I hope it stays a priority for the project.

Thanks for all the help with this bug. I am closing it now.
